python/opae.admin/opae/admin/tools/opaeuio.py

```diff
@@ -0,0 +1,185 @@
+                        help='release the given device from uio')
+    parser.add_argument('-d', '--driver', default='dfl-uio-pdev',
+                        help='driver to bind on init')
+    parser.add_argument('-u', '--user', default='root',
+                        help='userid to assign during init')
+    parser.add_argument('-g', '--group', default='root',
+                        help='groupid to assign during init')
+    parser.add_argument('-v', '--version', action='version',
+                        version='%(prog)s {}'.format(OPAEUIO_VERSION),
+                        help='display version information and exit')
+    return parser, parser.parse_args()
+
+
+def load_driver(driver):
+    """Load a Linux kernel module via modprobe.
+
+    driver - name of the driver to load.
+    """
+    return subprocess.call(['modprobe', driver])
+
+
+def initialize_uio(device, pac, new_owner, driver):
+    """Bind a DFL device and prepare it for use with uio.
+
+    device - the DFL device name.
+    pac - the dfl object (from dfl.enum()) corresponding to device.
+    new_owner - user:group for the owner of the resulting uio device.
+    driver - the new uio driver to bind.
+
+    Unbind the given DFL device from its current driver, re-assigning it
+    to the driver named by the last parameter. Discover the newly-created
+    uio device, and set the device permissions to rw-rw---- for new_owner,
+    given that new_owner != 'root:root'.
+    """
+    if pac.driver_override.value == driver:
+        print('{} is already bound to {}.'.format(device, driver))
+        return
+
+    print('Unbinding {} from {}.'.format(device, pac.driver.name))
+    pac.unbind()
+
+    print('Setting driver override to {}.'.format(driver))
+    pac.driver_override.value = driver
+
+    print('Loading {}.'.format(driver))
+    load_driver(driver)
+
+    print('Probing {}..'.format(device))
+    with open('/sys/bus/dfl/drivers_probe', 'w') as fp:
+        fp.write(device)
+
+    time.sleep(0.25)
+
+    uio_device = pac.find_one(os.path.join('uio_pdrv_genirq.*.auto',
+                                           'uio',
+                                           'uio*'))
+    if uio_device:
+        uio_device = os.path.join('/dev', uio_device.name)
+    else:
+        print('Error discovering char device for {}'.format(device))
+        sys.exit(1)
+    print('UIO device is {}.'.format(uio_device))
+
+    if new_owner != 'root:root':
+        items = new_owner.split(':')
+        user = pwd.getpwnam(items[0]).pw_uid
+        group = -1
+        if len(items) > 1:
+            group = grp.getgrnam(items[1]).gr_gid
+            print('Assigning {} to {}:{}'.format(uio_device, items[0], items[1]))
+        else:
+            print('Assigning {} to {}'.format(uio_device, items[0]))
+        os.chown(uio_device, user, group)
+        print('Changing permissions for {} to rw-rw----'.format(uio_device))
+        os.chmod(uio_device, 0o660)
```

Comment:
chmod 660

Reply:
This is how you write octal constants in Python 3.